src/viewing.jl

```diff
@@ -73,6 +73,19 @@ Return the indices of the `domain` that are inside the `geometry`.
 """
 indices(domain::Domain, geometry::Geometry) = filter(i -> domain[i] ⊆ geometry, 1:nelements(domain))
 
+function indices(grid::Grid{2}, poly::Polygon{2})
+  dims = size(grid)
+  mask = zeros(Int, dims)
```

Comment:
use spzeros to avoid allocating the entire mask, it is already imported from SparseArrays